### DISCUSSION TITLE
A way to hard set the mic and speaker option

### DISCUSSION THREAD

**Discussion by tan-banana** [vivekuppal/transcribe#191] [Ideas]
I need a way to set the specific mic and speaker not just the way they come up on the list like -1,0,1,2 position in the current code. It changes places if you use the current code and add a new speaker to the computer.

**Comment by vivekuppal** [comment]
This is a good idea.
The best way to achieve this would be to recognize the speaker, mic by name as it shows up in the list.

This can be achieved by altering the method `MicRecorder::set_device` and `SpeakerRecorder::set_device`

https://github.com/vivekuppal/transcribe/blob/33d69bdd162a97cd321e84c6471d9cb0b26ed072/sdk/audio_recorder.py#L217
https://github.com/vivekuppal/transcribe/blob/33d69bdd162a97cd321e84c6471d9cb0b26ed072/sdk/audio_recorder.py#L275
